packages/vite/src/node/config.ts

```diff
@@ -1766,7 +1766,7 @@ async function bundleConfigFile(
 
           // externalize bare imports
           build.onResolve(
-            { filter: /^[^.].*/ },
+            { filter: /^[^.#].*/ },
             async ({ path: id, importer, kind }) => {
```

Comment:
`resolveByViteResolver` uses `tryNodeResolve` and `tryNodeResolve` does not resolve imports fields. Resolving imports field in `resolveByViteResolver` can also fix the reported issue. But typescript files referenced by imports field doesn't work with that fix (I guess this didn't work from before). I made imports field to be bundled instead to allow typescript files to be referenced by imports field.